### PR TITLE
Atc external url

### PR DESCRIPTION
**Pull request by zachgersh**
Fixes #176

- You can now have ATC_EXTERNAL_URL appear in your container
- For some reason a lot of fakes were missing. WUT
- Any problems, speak to @utako

### CODE REVIEW

**Comment by xoebus** [comment]
Can you run `go fmt` on the code and squash please? All the fakes have become misaligned so the diff is huge.

File: engine/exec_engine_hooks_test.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ var _ = Describe("Exec Engine With Hooks", func() {
 		fakeDelegateFactory = new(fakes.FakeBuildDelegateFactory)
 		fakeDB = new(fakes.FakeEngineDB)
 
-		execEngine = engine.NewExecEngine(fakeFactory, fakeDelegateFactory, fakeDB)
+		execEngine = engine.NewExecEngine(fakeFactory, fakeDelegateFactory, fakeDB, "http://google.com")
```
Comment (xoebus) [inline engine/exec_engine_hooks_test.go]: Please use `http://example.com` instead of Google.

File: engine/exec_engine_test.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ var _ = Describe("ExecEngine", func() {
 		fakeDelegateFactory = new(fakes.FakeBuildDelegateFactory)
 		fakeDB = new(fakes.FakeEngineDB)
 
-		execEngine = engine.NewExecEngine(fakeFactory, fakeDelegateFactory, fakeDB)
+		execEngine = engine.NewExecEngine(fakeFactory, fakeDelegateFactory, fakeDB, "http://google.com")
```
Comment (xoebus) [inline engine/exec_engine_test.go]: Same here.

File: engine/exec_engine_try_test.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ var _ = Describe("Exec Engine with Try", func() {
 		fakeDelegateFactory = new(fakes.FakeBuildDelegateFactory)
 		fakeDB = new(fakes.FakeEngineDB)
 
-		execEngine = engine.NewExecEngine(fakeFactory, fakeDelegateFactory, fakeDB)
+		execEngine = engine.NewExecEngine(fakeFactory, fakeDelegateFactory, fakeDB, "http://google.com")
```
Comment (xoebus) [inline engine/exec_engine_try_test.go]: Also here.

File: engine/exec_engine_try_test.go [hunk 4]
```diff
@@ -54,6 +54,7 @@ var _ = Describe("Exec Engine with Try", func() {
 			BuildName:    "42",
 			JobName:      "some-job",
 			PipelineName: "some-pipeline",
+			ExternalURL:  "http://google.com",
```
Comment (xoebus) [inline engine/exec_engine_try_test.go]: etc.

**Comment by xoebus** [comment]
Looks good! Left some comments above but this should be good to merge afterwards.

File: worker/fakes/fake_container.go [hunk 5]
```diff
@@ -473,6 +481,38 @@ func (fake *FakeContainer) CurrentCPULimitsReturns(result1 garden.CPULimits, res
 	}{result1, result2}
 }
 
+func (fake *FakeContainer) LimitDisk(limits garden.DiskLimits) error {
```
Comment (vito) [inline worker/fakes/fake_container.go]: These shouldn't be re-added; make sure your local checkout of `cloudfoundry-incubator/garden` is the latest.

**Comment by zachgersh** [comment]
@vito / @xoebus thanks for the review. Should be fixed now.  If you'd like me to change the external-url flag description or set it to required before merging - just ping me.

**Comment by vito** [comment]
thx

**Comment by xoebus** [comment]
ya

**Pull request merged**